From 9c2c33fb10531c88c67f7cdf2cb32606d8c13aa7 Mon Sep 17 00:00:00 2001
From: Xing Zheng <zhengxing@rock-chips.com>
Date: Fri, 15 May 2020 10:00:34 +0800
Subject: [PATCH] ASoC: rk3308_codec: enable DAC HPOUT_POP_SOUND LINEOUT

We need to keep HPOUT_POP_SOUND enabled and powered VCM.

Reported-by: XiaoTan Luo <lxt@rock-chips.com>
Signed-off-by: Xing Zheng <zhengxing@rock-chips.com>
Change-Id: I9cd5151dfa6f791985cd381616674df0348e2fcb
---
 sound/soc/codecs/rk3308_codec.c | 17 +++++++----------
 1 file changed, 7 insertions(+), 10 deletions(-)

diff --git a/sound/soc/codecs/rk3308_codec.c b/sound/soc/codecs/rk3308_codec.c
index d1213b954d14..03eceaa1058f 100644
--- a/sound/soc/codecs/rk3308_codec.c
+++ b/sound/soc/codecs/rk3308_codec.c
@@ -2066,17 +2066,14 @@ static int rk3308_codec_dac_enable(struct rk3308_codec_priv *rk3308)
 	/* Waiting the stable reference voltage */
 	mdelay(1);
 
-	if (rk3308->dac_output == DAC_HPOUT ||
-	    rk3308->dac_output == DAC_LINEOUT_HPOUT) {
-		/* Step 03 */
-		regmap_update_bits(rk3308->regmap, RK3308_DAC_ANA_CON01,
-				   RK3308_DAC_HPOUT_POP_SOUND_L_MSK |
-				   RK3308_DAC_HPOUT_POP_SOUND_R_MSK,
-				   RK3308_DAC_HPOUT_POP_SOUND_L_WORK |
-				   RK3308_DAC_HPOUT_POP_SOUND_R_WORK);
+	/* Step 03 */
+	regmap_update_bits(rk3308->regmap, RK3308_DAC_ANA_CON01,
+			   RK3308_DAC_HPOUT_POP_SOUND_L_MSK |
+			   RK3308_DAC_HPOUT_POP_SOUND_R_MSK,
+			   RK3308_DAC_HPOUT_POP_SOUND_L_WORK |
+			   RK3308_DAC_HPOUT_POP_SOUND_R_WORK);
 
-		udelay(20);
-	}
+	udelay(20);
 
 	if (rk3308->codec_ver == ACODEC_VERSION_B &&
 	    (rk3308->dac_output == DAC_LINEOUT ||
-- 
2.35.3

